addons/sale/models/sale_order.py

```diff
@@ -1695,6 +1781,101 @@ def _get_default_payment_link_values(self):
             'amount_paid': self.amount_paid,
         }
 
+    # EDI #
+
+    def create_document_from_attachment(self, attachment_ids):
+        """ Create the sale orders from given attachment_ids and redirect newly create order view.
+
+        :param list attachment_ids: List of attachments process.
+        :return: An action redirecting to related sale order view.
+        :rtype: dict
+        """
+        orders = self._create_order_from_attachment(attachment_ids)
+        action_vals = {
+            'name': _("Generated Orders"),
+            'domain': [('id', 'in', orders.ids)],
+            'res_model': 'sale.order',
+            'type': 'ir.actions.act_window',
+            'context': self._context
+        }
+        if len(orders) == 1:
+            action_vals.update({
+                'views': [[False, 'form']],
+                'res_id': orders.id,
+            })
+        else:
+            action_vals.update({
+                'views': [[False, 'list'], [False, 'kanban'], [False, 'form']],
+            })
+
+        return action_vals
```

Comment:
Check out `odoo/odoo/addons/base/models/ir_ui_view.py`, there is some `Model._get_records_action` 

![immagine](https://github.com/user-attachments/assets/bf8d0031-77a2-4abe-b88b-509c1d96708e)

Reply:
Got it!